pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -669,3 +679,31 @@ func createAppInstances(ctxPtr *zedagentContext,
 	ctxPtr.getconfigCtx.subAppInstanceStatus.Iterate(
 		addAppInstanceFunc)
 }
+
+func getSMARTAttributeValue(field string) int64 {
+	filename := "/run/SMART_details"
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		log.Errorf("getSMARTAttributeValue: exception while opening %s. %s", filename, err.Error())
+		return -1
+	}
+	strData := string(data)
+	if strings.Contains(strData, "SMART support is: Enabled") {
+		log.Functionf("getSMARTAttributeValue: SMART support is available on the device. Fetching %s", field)
+		pattern := regexp.MustCompile("[0-9 ]*" + field + "[^\n]*")
+		result := strings.FieldsFunc(pattern.FindString(strData), func(c rune) bool { return c == ' ' })
+		if len(result) > 0 {
+			if powerCycle, err := strconv.ParseInt(result[len(result)-1], 10, 64); err != nil {
+				log.Errorf("getSMARTAttributeValue: exception while parsing SMART output %v. %s", result, err.Error())
```

Comment:
I don't think we should error just because some disk/ssd controller doesn't support the power cycle counter. Just set counter to -1.

Reply:
Changed it to Warning.